### PR TITLE
split the crashlooping pod test to more precisely tag recycler pods

### PR DESCRIPTION
**Pull request by deads2k**
it looks like we have a significant number of recycler pod problems compared against "other" pods.  Both are problematic, but recycler pods are less bad and specifically storage related. This refactors the test and allows filtering.

Review with whitespace off

### CODE REVIEW

File: test/extended/operators/cluster.go [hunk 1]
```diff
@@ -16,105 +16,140 @@ import (
...
+	return strings.HasPrefix(pod.Namespace, "openshift-") || strings.HasPrefix(pod.Namespace, "kube-")
+}
+
+func not(filterFn podFilter) podFilter {
+	return func(pod *corev1.Pod) bool {
+		return !filterFn(pod)
+	}
+}
+
+func recyclerPod(pod *corev1.Pod) bool {
```
Comment (mfojtik) [inline test/extended/operators/cluster.go:39]: nit: `isRecyclerPod`

Comment (mfojtik) [reply test/extended/operators/cluster.go:39]: actually nvmd, usage reads fine

**Comment by deads2k** [comment]
refining signal for e2e failures.  passed in the job that was impacted.

**Pull request merged**